### PR TITLE
add code unit tests for the module plugins/flexibleengine

### LINKED ISSUE #517: Add code unit tests for the module plugins/flexibleengine
I will take this

### PR DESCRIPTION
**Pull request by baixinsui**
fixed #517

### CODE REVIEW

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/utils/RetryTemplateServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        final KeystoneListProjectsResponse result = retryTemplateService.queryProjectInfo(httpRequestBase);
+
+        // Verify the results
+        Assertions.assertFalse(Objects.isNull(result));
+        Assertions.assertEquals(1, result.getProjects().size());
+        Assertions.assertEquals(projectId, result.getProjects().get(0).getId());
+    }
+
+    @Test
+    void testQueryMetricData_cpu_usage() throws URISyntaxException {
```
Comment (swaroopar) [inline plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/utils/RetryTemplateServiceTest.java:60]: method names without underscore pls.

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/constant/FlexibleEngineMonitorConstantsTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.eclipse.xpanse.plugins.flexibleengine.monitor.constant;
+
+import org.junit.jupiter.api.BeforeEach;
+
+class FlexibleEngineMonitorConstantsTest {
+
+    private FlexibleEngineMonitorConstants flexibleEngineMonitorConstantsUnderTest;
+
+    @BeforeEach
```
Comment (swaroopar) [inline plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/constant/FlexibleEngineMonitorConstantsTest.java:9]: Do we need this empty test?

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/models/FlexibleEngineResourcePropertyTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import org.eclipse.xpanse.modules.models.service.deploy.enums.DeployResourceKind;
+import org.junit.jupiter.api.Test;
+
+class FlexibleEngineResourcePropertyTest {
+
+    @Test
+    void testGetVmProperties_VM() {
```
Comment (swaroopar) [inline plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/models/FlexibleEngineResourcePropertyTest.java:13]: same here. Method names only regular java style camel case pls.

**Review by swaroopar — CHANGES_REQUESTED** [review]
added a comment regarding naming format. Rest looks great.

File: plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/utils/MetricsService.java [hunk 4]
```diff
@@ -282,18 +275,36 @@ public List<Metric> getMetricsForService(ServiceMetricRequest serviceMetricReque
...
+    private Project getProjectInfoByRegion(AbstractCredentialInfo credential, String region) {
+
+        Project project = null;
+        if (StringUtils.isNotBlank(region)) {
+            String projectQueryUrl =
+                    flexibleEngineMonitorConverter.buildProjectQueryUrl(region).toString();
+            project =
+                    queryProjectInfo(credential, projectQueryUrl);
+        }
+        if (Objects.isNull(project) || StringUtils.isBlank(project.getId())) {
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/utils/MetricsService.java:287]: @sonatype-lift ignoreall

File: plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/constant/FlexibleEngineMonitorConstantsTest.java (hunk 2 above)
Comment (baixinsui) [reply plugins/flexibleengine/src/test/java/org/eclipse/xpanse/plugins/flexibleengine/monitor/constant/FlexibleEngineMonitorConstantsTest.java]: I removed it.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**